src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -276,35 +281,80 @@ export class ObjectExplorerService implements IObjectExplorerService {
 	private async handleSessionCreated(session: azdata.ObjectExplorerSession): Promise<void> {
 		let connection: ConnectionProfile | undefined = undefined;
 		let errorMessage: string | undefined = undefined;
-		if (this._sessions[session.sessionId!]) {
-			connection = this._sessions[session.sessionId!].connection;
+		const sessionId = session.sessionId;
 
-			try {
-				if (session.success && session.rootNode) {
-					let server = this.toTreeNode(session.rootNode, undefined);
-					server.connection = connection;
-					server.session = session;
-					this._activeObjectExplorerNodes[connection!.id] = server;
+
+		await new Promise<void>((resolve, reject) => {
+			const cleanup = () => {
+				if (connection) {
+					this._connectionsWaitingForSession.delete(connection.id);
 				}
-				else {
-					errorMessage = session && session.errorMessage ? session.errorMessage : errSessionCreateFailed;
-					this.logService.error(errorMessage);
+				createNewSessionListener.dispose();
+				clearTimeout(timeout);
+			}
+			const onTimeout = () => {
+				if (!this._sessions[sessionId]) {
```

Comment:
Do we actually need separate errors here? Seems like the stuff in the else is all we need

Reply:
Usually, this error won't happen unless the providers are misbehaving. Having separate errors will just help us figure out what exactly is happening in the OE service.

Reply:
Please update the messages to be clearer though - right now they don't really indicate the problem at all. For example, you always have a sessionID so should always be including that. And then neither branch does anything with the actual session or lack thereof - the second adds some info about the connection but isn't that also available in both cases as well?

Reply:
@Charles-Gagnon , updated the error message to include more context.